warehouse/source/http.go

```diff
@@ -0,0 +1,188 @@
+package source
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+	"strings"
+
+	"github.com/rudderlabs/rudder-server/warehouse/internal/model"
+
+	"github.com/rudderlabs/rudder-server/services/notifier"
+
+	ierrors "github.com/rudderlabs/rudder-server/warehouse/internal/errors"
+	lf "github.com/rudderlabs/rudder-server/warehouse/logfield"
+
+	"github.com/samber/lo"
+)
+
+// InsertJobHandler adds a job to the warehouse_jobs table
+func (m *Manager) InsertJobHandler(w http.ResponseWriter, r *http.Request) {
+	defer func() { _ = r.Body.Close() }()
+
+	var payload insertJobRequest
+	if err := json.NewDecoder(r.Body).Decode(&payload); err != nil {
+		m.logger.Warnw("invalid JSON in request body for inserting source jobs", lf.Error, err.Error())
+		http.Error(w, ierrors.ErrInvalidJSONRequestBody.Error(), http.StatusBadRequest)
+		return
+	}
+
+	if err := validatePayload(&payload); err != nil {
+		m.logger.Warnw("invalid payload for inserting source job", lf.Error, err.Error())
+		http.Error(w, fmt.Sprintf("invalid payload: %s", err.Error()), http.StatusBadRequest)
+		return
+	}
+
+	tableUploads, err := m.tableUploadsRepo.GetByJobRunTaskRun(
+		r.Context(),
+		payload.SourceID,
+		payload.DestinationID,
+		payload.JobRunID,
+		payload.TaskRunID,
+	)
+	if err != nil {
+		if errors.Is(r.Context().Err(), context.Canceled) {
+			http.Error(w, ierrors.ErrRequestCancelled.Error(), http.StatusBadRequest)
+			return
+		}
+		m.logger.Errorw("extracting tableNames for inserting source job", lf.Error, err.Error())
+		http.Error(w, "can't extract tableNames", http.StatusInternalServerError)
+		return
+	}
+
+	tableNames := lo.Map(tableUploads, func(item model.TableUpload, index int) string {
+		return item.TableName
+	})
+	tableNames = lo.Filter(lo.Uniq(tableNames), func(tableName string, i int) bool {
+		switch strings.ToLower(tableName) {
+		case "rudder_discards", "rudder_identity_mappings", "rudder_identity_merge_rules":
```

Comment:
Don't we have enums for these? :thinking:

Reply:
We have constants for these:
```GO
	tableNames = lo.Filter(lo.Uniq(tableNames), func(tableName string, i int) bool {
		switch strings.ToLower(tableName) {
		case whutils.DiscardsTable, whutils.IdentityMappingsTable, whutils.IdentityMergeRulesTable:
			return false
		default:
			return true
		}
	})
```